src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -2548,6 +2647,326 @@ public final void addBatch() throws SQLServerException {
+        localUserSQL = localUserSQL.trim();
+        
+        // ignore all comments
+        if (checkAndRemoveComments()) {
+            return parseUserSQLForColumnListDW();
+        }
+        
+        //check if optional column list was provided
+        // Columns can have the form of c1, [c1] or "c1". It can escape ] or " by ]] or "".
+        if (localUserSQL.substring(0, 1).equalsIgnoreCase("(")) {
+            localUserSQL = localUserSQL.substring(1);
+            return parseUserSQLForColumnListDWHelper(new ArrayList<String>());
+        }
+        return null;
+    }
+
+    private ArrayList<String> parseUserSQLForColumnListDWHelper(ArrayList<String> listOfColumns) {
+        localUserSQL = localUserSQL.trim();
+        
+        // ignore all comments
+        if (checkAndRemoveComments()) {
+            return parseUserSQLForColumnListDWHelper(listOfColumns);
+        }
+        
+        if (localUserSQL.charAt(0) == ')') {
+            localUserSQL = localUserSQL.substring(1);
+            return listOfColumns;
+        }
+        
+        if (localUserSQL.charAt(0) == ',') {
+            localUserSQL = localUserSQL.substring(1);
+            return parseUserSQLForColumnListDWHelper(listOfColumns);
+        }
+        
+        if (localUserSQL.charAt(0) == '[') {
+            int tempint = localUserSQL.indexOf("]", 1);
+            
+            // keep checking if it's escaped
+            while (localUserSQL.charAt(tempint + 1) == ']') {
+                localUserSQL = localUserSQL.substring(0, tempint) + localUserSQL.substring(tempint + 1);
+                tempint = localUserSQL.indexOf("]", tempint + 1);
+            }
+            
+            // we've found a ] that is actually trying to close the square bracket.
+            String tempstr = localUserSQL.substring(1, tempint);
+            localUserSQL = localUserSQL.substring(tempint + 1);
+            listOfColumns.add(tempstr);
+            return parseUserSQLForColumnListDWHelper(listOfColumns);
+        }
+        
+        if (localUserSQL.charAt(0) == '\"') {
+            int tempint = localUserSQL.indexOf("\"", 1);
+            
+            // keep checking if it's escaped
+            while (localUserSQL.charAt(tempint + 1) == '\"') {
+                localUserSQL = localUserSQL.substring(0, tempint) + localUserSQL.substring(tempint + 1);
+                tempint = localUserSQL.indexOf("\"", tempint + 1);
+            }
+            
+            // we've found a " that is actually trying to close the quote.
+            String tempstr = localUserSQL.substring(1, tempint);
+            localUserSQL = localUserSQL.substring(tempint + 1);
+            listOfColumns.add(tempstr);
+            return parseUserSQLForColumnListDWHelper(listOfColumns);
+        }
+
+        // At this point, the next chunk of string is the column name, without starting with [ or ".
+        StringBuilder sb = new StringBuilder();
+        while (localUserSQL.length() > 0) {
```

Comment:
No need for nested if blocks here, this is simpler. 

```
if (localUserSQL.charAt(0) == ',') {
}
else if (localUserSQL.charAt(0) == ')'){
}
} else if (checkAndRemoveComments()) {
}
else{
}

```

Reply:
done.